src/dhcpv6-ia.c

```diff
@@ -1364,7 +1364,17 @@ ssize_t dhcpv6_ia_handle_IAs(uint8_t *buf, size_t buflen, struct interface *ifac
 						a->clid_len = clid_len;
 						memcpy(a->clid_data, clid_data, clid_len);
 						a->iaid = ia->iaid;
-						a->length = reqlen;
+						if (is_pd) {
+							if (reqlen < iface->dhcpv6_pd_min) {
+								syslog(LOG_INFO, "clamping requested PD from %d to %d",
+								       reqlen, iface->dhcpv6_pd_min);
+								a->length = iface->dhcpv6_pd_min;
+							} else {
+								a->length = reqlen;
+							}
+						} else {
+							a->length = reqlen;
+						}
```

Comment:
Can you restructure this to something better like

```suggestion
						a->length = reqlen;
						if (is_pd && reqlen < iface->dhcpv6_pd_min) {
							syslog(LOG_INFO, "clamping requested PD from %d to %d",
								    reqlen, iface->dhcpv6_pd_min);
							a->length = iface->dhcpv6_pd_min;
						}
```